cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -144,11 +144,15 @@ public AbstractAppFabricHttpHandler(Authenticator authenticator) {
   }
 
   protected int getInstances(HttpRequest request) throws IOException, NumberFormatException {
-    return parseBody(request, Instances.class).getInstances();
+    Instances instances = parseBody(request, Instances.class);
```

Comment:
this looks like an unrelated change. I have noticed this warning myself and was tempted to fix it also...
If you want to fix this, you should throw an exception that can be recognized by all callers, so that they can return BAD_REQUEST. And make that change in all callers, too :)